### ISSUE TITLE
Documentation Issue - Continuous Testing - Docker - Bare Minimum Config settings

### ISSUE THREAD

**Issue by etlnibn** [sitespeedio/sitespeed.io#2708]
Continuous Testing - Docker - For a Newbie.
The issue was first raised on Slack and I was requested to log an issue to address the documentation.

- I am quite new to sitespeed.io generally but I have been able to get things going very easily (up to a point) thanks to the wonderful dockers which are available
- I can run tests manually and everything appears as expected in the dashboards and now I want to move to continuous testing and have read the documentation many times and still feel totally confused.
- The documentation in this areas seems to be written more like a record of how the solution is being used internally for the sitespeed.io team rather than a guide for how to get up and running - and I am finding it difficult to connect the dots.
- I have forked the repository, I have updated the test URLs and I know that I want to mainly focus on simple emulated mobile on 4g (and possibly desktop) testing.
- Where I think it becomes really confusing is regarding the configuration. Would it be possible to extend the documentation for probably a lot of people who are running the dockers on a local Linux server and explain what exactly needs to be done with the config files.
- Perhaps most people are running on AWS and want call backs to Slack but for the "up and running in 5 minutes" people who just want to loop the current testing - it feels like a huge jump
- What is the bare minimum I need to change to get this working. What can I / should I remove - I can guess but it would be helpful with guidance
- Are other people running continuous testing from the dockers on a local linux server - can you share your config files (minus any secrets)- and anything else you needed to adjust

https://sitespeedio.slack.com/archives/C03DHTPSA/p1571386495051800

**Comment by soulgalore** [comment]
Yeah I see your point. There's a big gap between running the first time and setting up something like dashboard.sitespeed.io.

Let us work on that.

**Comment by etlnibn** [comment]
I would assume that if you are using the docker containers then there are lots of values which are already known and therefore moving to a repeating test scenario could be relatively straightforward . Yes, you need the list of URLs and to decide on whether you want to test desktop, mobile, WPT (or several) but could be a little less of a step up.

In the meantime I will keep reading, and testing. Many thanks.

**Comment by tentekal** [comment]
I'd like to voice my interest in this as well. I'm terribly confused about configuration--especially secrets.json. It's really not clear how we need to modify configs and graphite in order to get the data visualized in grafana. Additionally, as mentioned by etlnibn, we need documentation on a local linux setup and how to cross the corporate proxy easily (can this be done in the json config? It's unclear, seems it's only possible with CLI).

**Comment by soulgalore** [comment]
Ah sorry, I've been busy with thing with higher prio. I'll create the docs, but I have one documentation that are before in line, how do synthetic testing in general, when that is done, I'll can add setup docs in more detail.

A couple of things: Everything you can do in the CLI you can do with the configuration JSON file(s). They are the same, just different representation of the same data. We use [yargs](https://github.com/yargs/yargs) for that.

The secrets.json file is an example where we can have all configuration open at Github but still have API keys hidden, so that no one can use our setup. When you a configuration file, it can extend other file. A use was for you could be that you have a base JSON configuration file with all configuration that you have in common between different tests, and then you can have other files that extends that, where you add the specific configs to that run.

The simplest setup running a test, sending data to Graphite looks like this:
`sitespeed.io --graphite.host <my.host.com> https://www.sitespeed.io` 

In that example we test https://www.sitespeed.io with default configuration, sending the data to our Graphite instance at my.host.com.

The same in a JSON configuration looks like this: 
```JSON
"graphite": {
   "host": "<my.host.com>"
}
```

Name that file myconfiguration.json. Then run it like this:
`sitespeed.io  --config myconfiguration.json  https://www.sitespeed.io`

If you use Docker you can checkout https://www.sitespeed.io/documentation/sitespeed.io/docker/.

> local linux setup and how to cross the corporate proxy easily 

Running tests through a proxy is probably not a good idea, if the proxy somehow touches the traffic? Or are the idea that you need test as users inside of the proxy?

Best
Peter

**Comment by tentekal** [comment]
Thanks for the pointers, I figured most of this out when messing around in the environment (using the docker and the dashboard.sitespeed.io docker), just would have been helpful to have all that documented in one page or so.

I agree that through the proxy isn't ideal, but we get some pretty comparable stats from doing this, and presently a cloud-based deployment or anything outside our 'DMZ' isn't possible.
